client/app/states/designer/blueprints/editor/canvasCtrl.js

```diff
@@ -53,9 +55,60 @@
     };
 
     $scope.addNewNode = function(newNode) {
-      $scope.chartViewModel.addNode(newNode);
+      if (!newNode.type || newNode.type !== 'new-generic') {
+        getTags(newNode.id).then(function(tags) {
+          newNode.tags = tags;
+          $scope.chartViewModel.addNode(newNode);
+        }, function() {
+          $scope.chartViewModel.addNode(newNode);
+        });
+      } else {
+        newNode.tags = [];
+        $scope.chartViewModel.addNode(newNode);
+      }
     };
 
+    function getTags(id) {
+      var deferred = $q.defer();
+
+      var attributes = ['categorization', 'category.id', 'category.single_value'];
+      var options = {
+        expand: 'resources',
+        attributes: attributes,
+      };
+
+      var collection = "service_templates" + "\/" + id + "\/" + 'tags';
+
+      CollectionsApi.query(collection, options).then(loadSuccess, loadFailure);
+
+      function loadSuccess(response) {
+        var tags = [];
+        angular.forEach(response.resources, processTag);
+
+        function processTag(tag) {
+          tags.push(getSmTagObj(tag));
+        }
+
+        function getSmTagObj(tag) {
+          return {id: tag.id,
+            category: {id: tag.category.id},
+            categorization: {
+              display_name: tag.categorization.display_name,
+            }
+          };
+        }
+
+        deferred.resolve(tags);
+      }
+
+      function loadFailure() {
+        console.log('There was an error service template tags');
```

Comment:
How'd that get in there?!  I did a global search for all console.log's in my last PR.  I'll fix as well.
Thanks!